.github/workflows/test.yml

```diff
@@ -15,25 +15,128 @@ jobs:
           GITHUB_TOKEN: "${{ secrets.GITHUB_TOKEN }}"
     if: "!startsWith(github.ref, 'refs/tags/') && github.ref != 'refs/heads/master'"
 
-  test-coverage-upload:
+  split-test-files: 
     runs-on: ubuntu-latest
+    steps: 
+      - uses: actions/checkout@v2
+      - name: Create a file with all the pkgs
+        run: go list ./... > pkgs.txt
+      - name: Split pkgs into 4 files
+        run: split -n l/4 --additional-suffix=.txt ./pkgs.txt
+      # cache multiple
+      - uses: actions/upload-artifact@v2
+        with:
+          name: '${{ github.sha }}-aa'
+          path: ./xaa.txt
```

Comment:
can we use a more descriptive name?

Reply:
why? this won't affect anyone. this is the auto output from the `split` cmd

Reply:
```
alessio@bangalter:~/work/cosmos-sdk$ split --help
Usage: split [OPTION]... [FILE [PREFIX]]
Output pieces of FILE to PREFIXaa, PREFIXab, ...;
default size is 1000 lines, and default PREFIX is 'x'.
```

You could try customise output file prefix - though I agree it's not critical